### PR TITLE
fix get similar incidents query bug

### PR DESCRIPTION
**Pull request by sabaghadar**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://github.com/demisto/etc/issues/37557

## Description
A few sentences describing the overall goals of the pull request's commits.

## Screenshots
Paste here any images that will help the reviewer

## Minimum version of Cortex XSOAR
- [x] 5.5.0
- [ ] 6.0.0
- [ ] 6.1.0
- [ ] 6.2.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [ ] No

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

File: Packs/Base/Scripts/FindSimilarIncidentsByText/find_similar_incidents_text_test.py [hunk 1]
```diff
@@ -1,3 +1,5 @@
+import json
```
Comment (roysagi) [inline Packs/Base/Scripts/FindSimilarIncidentsByText/find_similar_incidents_text_test.py:1]: change the file name to:
FindSimilarIncidentsByText_test.py

File: Packs/Base/Scripts/FindSimilarIncidentsByText/FindSimilarIncidentsByText.py [hunk 2]
```diff
@@ -50,23 +50,26 @@ def get_incidents_by_time(incident_time, incident_type, incident_id, hours_time_
...
-    res = demisto.executeCommand("getIncidents",
-                                 {'query': query,
-                                  'size': max_number_of_results, 'sort': '%s.desc' % time_field})
 
-    if res[0]['Type'] == entryTypes['error']:
-        raise Exception(str(res[0]['Contents']))
+    res = demisto.executeCommand('GetIncidentsByQuery', {
+        'query': query,
+        'fromDate': min_date.isoformat(),
+        'toDate': max_date.isoformat(),
```
Comment (roysagi) [inline Packs/Base/Scripts/FindSimilarIncidentsByText/FindSimilarIncidentsByText.py:65]: `isoformat` returns the time string in the following format:
image

Please let's make sure this format is good - if I recall correctly we had an issue with the millisecond info - @bakatzir am I right here?

**Review by roysagi — CHANGES_REQUESTED** [review]

**Comment by roysagi** [comment]
@erezh31 - This script uses an old docker image named [machine-learning](https://hub.docker.com/r/demisto/machine-learning) do you know by any means if we can change it to use the newer [ml](https://hub.docker.com/r/demisto/ml)?

**Comment by erezh31** [comment]
Yes, demisto/sklearn should be good for this script - please validate that

**Comment by sabaghadar** [comment]
> Yes, demisto/sklearn should be good for this script - please validate that

I got some errors (related to python version i think) when using demisto/sklearn. @erezh31 should i keep the changes or use a different docker image? ( for example: Packs/Base/Scripts/FindSimilarIncidentsByText/FindSimilarIncidentsByText.py :33, 34 : using + operand on dict_items)

File: Packs/Base/Scripts/FindSimilarIncidentsByText/FindSimilarIncidentsByText.py (hunk 2 above)
Comment (sabaghadar) [reply Packs/Base/Scripts/FindSimilarIncidentsByText/FindSimilarIncidentsByText.py:65]: looks like the script we are using here is converting the date to this format anyway -
https://github.com/demisto/content/blob/master/Packs/Base/Scripts/GetIncidentsByQuery/GetIncidentsByQuery.py#L125-L140

File: Packs/Base/ReleaseNotes/1_11_9.md [hunk 3]
```diff
@@ -0,0 +1,4 @@
+
+#### Scripts
+##### FindSimilarIncidentsByText
+Fixed query in get_incidents_by_time
```
Comment (roysagi) [inline Packs/Base/ReleaseNotes/1_11_9.md:4]: please write something that starts with "Fixed an issue where..."
Think of this from the customer's perspective what did he see that did not work and now is working.

File: Packs/Base/Scripts/FindSimilarIncidentsByText/FindSimilarIncidentsByText.yml [hunk 4]
```diff
@@ -106,7 +106,7 @@ tags:
 - incidents
 timeout: '0'
 type: python
-dockerimage: demisto/machine-learning
+dockerimage: demisto/sklearn:1.0.0.19770
```
Comment (roysagi) [inline Packs/Base/Scripts/FindSimilarIncidentsByText/FindSimilarIncidentsByText.yml:109]: please change the subtype here to `python3`

**Review by roysagi — APPROVED** [review]
Please make the subtype change and we are good to go.

**Comment by sabaghadar** [comment]
Tested the command with the changes on dev6.

**Comment by Itay4** [comment]
build fails on approved python version change cause of docker image update
merging

**Pull request merged**